File: backend/pomoro-do/src/main/java/com/tico/pomoro_do/domain/category/dto/response/CategoryInvitationResponse.java
package com.tico.pomoro_do.domain.category.dto.response;

import lombok.Builder;
import lombok.Getter;

@Getter
public class CategoryInvitationResponse {

    private final Long categoryInvitationId;
    private final String categoryName;
    private final String ownerNickname;

    @Builder
    public CategoryInvitationResponse(Long categoryInvitationId, String categoryName, String ownerNickname) {
        this.categoryInvitationId = categoryInvitationId;
        this.categoryName = categoryName;
        this.ownerNickname = ownerNickname;
    }
}

File: Mobile/PomoroDo/app/src/main/java/com/tico/pomorodo/data/remote/models/response/category/CategoryInvitationResponse.kt
package com.tico.pomorodo.data.remote.models.response.category

import com.tico.pomorodo.data.model.CategoryInvitation
import kotlinx.serialization.Serializable

@Serializable
data class CategoryInvitationResponse(
    val categoryInvitationId: Int,
    val categoryName: String,
    val ownerNickname: String
)

fun CategoryInvitationResponse.toCategoryInvitation() = CategoryInvitation(
    categoryInvitationId = categoryInvitationId,
    categoryName = categoryName,
    ownerNickname = ownerNickname
)